cmd/snap-update-ns/bootstrap.c

```diff
@@ -75,6 +76,48 @@ setns_into_snap(const char* snap_name)
     return err;
 }
 
+static int
+switch_to_privileged_user()
+{
+    uid_t real_uid;
+
+    real_uid = getuid();
+    if (real_uid == 0) {
+        // We're running as root: no need to switch IDs
+        return 0;
+    }
+
+    struct __user_cap_header_struct hdr = { _LINUX_CAPABILITY_VERSION_3, 0 };
+    struct __user_cap_data_struct data[2] = { { 0 } };
+
+    data[0].effective = CAP_TO_MASK (CAP_SYS_ADMIN);;
+    data[0].permitted = data[0].effective;
+    data[0].inheritable = 0;
+    data[1].effective = 0;
+    data[1].permitted = 0;
+    data[1].inheritable = 0;
+
+    if (capset(&hdr, data) != 0) {
+        bootstrap_errno = errno;
+        bootstrap_msg = "cannot set permitted capabilities mask";
+        return -1;
+    }
+
+    if (seteuid(real_uid) != 0) {
```

Comment:
We're dropping the uid but never re-raising. We should permanently drop supplementary groups, gid and uid, no?